Genrate Client using OpenAPI spec tool, updated package.md and module.md (#3)
ballerina-platform/module-ballerinax-hubspot.marketing.transactional: README.md

> # Ballerina HubSpot Marketing Transactional Connector 

Reviewer: ```suggestion
# Ballerina HubSpot Marketing Transactional Emails Connector 
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

[![Trivy](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/actions/workflows/trivy-scan.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/actions/workflows/trivy-scan.yml)
[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/actions/workflows/build-with-bal-test-graalvm.yml)
[![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-hubspot.marketing.transactional.svg)](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/commits/master)
[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/hubspot.marketing.transactional.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%hubspot.marketing.transactional)

> # Overview

Reviewer: ```suggestion
## Overview
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

"[HubSpot](https://www.hubspot.com/) is an AI-powered customer relationship management (CRM) platform. 

> The ``ballerinax/hubspot.marketing.`transactional`` offers APIs to connect and interact with the [Hubspot Marketing Transactional Emails API](https://developers.hubspot.com/docs/reference/api/marketing/emails/transactional-emails) endpoints, specifically based on the [HubSpot REST API](https://developers.hubspot.com/docs/reference/api/overview)"

Reviewer: ```suggestion
[HubSpot](https://www.hubspot.com/) is an AI-powered customer relationship management (CRM) platform. 

The `ballerinax/hubspot.marketing.transactional` connector offers APIs to connect and interact with the [Hubspot Marketing Transactional Emails API](https://developers.hubspot.com/docs/reference/api/marketing/emails/transactional-emails) endpoints, specifically based on the [HubSpot REST API](https://developers.hubspot.com/docs/reference/api/overview)
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

## Setup guide

To use the HubSpot Marketing Transactional Emails connector, you must have access to the HubSpot API through a HubSpot developer account and a HubSpot App under it. Therefore you need to register for a developer account at HubSpot if you don't have one already.

### Step 1: Create/Login to a HubSpot Developer Account

If you have an account already, go to the [HubSpot developer portal](https://app.hubspot.com/)

If you don't have a HubSpot Developer Account you can sign up to a free account [here](https://developers.hubspot.com/get-started)

> ### Step 2 (Optional): Create a [Developer Test Account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account

Reviewer: ```suggestion
### Step 2 (Optional): Create a Developer Test Account
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

> Within app developer accounts, you can create developer test accounts to test apps and integrations without affecting any real HubSpot data.

Reviewer: ```suggestion
Within app developer accounts, you can create a [developer test account](https://developers.hubspot.com/beta-docs/getting-started/account-types#developer-test-accounts) under your account to test apps and integrations without affecting any real HubSpot data.
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

> **Note: These accounts are only for development and testing purposes. In production you should not use Developer Test Accounts.**

Reviewer: ```suggestion
> **Note:** These accounts are only for development and testing purposes. In production you should not use Developer Test Accounts.
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

   ![Hubspot create app final](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/main/docs/setup/resources/create_app_final.png)

### Step 5: Get your Client ID and Client Secret

- Navigate to the Auth section of your app. Make sure to save the provided Client ID and Client Secret.  
  ![Hubspot get credentials](https://raw.githubusercontent.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/main/docs/setup/resources/get_credentials.png)

### Step 6: Setup Authentication Flow

Before proceeding with the Quickstart, ensure you have obtained the Access Token using the following steps:

1. Create an authorization URL using the following format:

   ```
   https://app.hubspot.com/oauth/authorize?client_id=<YOUR_CLIENT_ID>&scope=<YOUR_SCOPES>&redirect_uri=<YOUR_REDIRECT_URI>
   ```

   Replace the `<YOUR_CLIENT_ID>`, `<YOUR_REDIRECT_URI>` and `<YOUR_SCOPES>` with your specific value.

2. Paste it in the browser and select your developer test account to intall the app when prompted.
3. A code will be displayed in the browser. Copy the code.

   ```
   Received code: na1-129d-860c-xxxx-xxxx-xxxxxxxxxxxx

>    ```

Reviewer: ```suggestion
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

   - Ballerina Program

   ```ballerina
      import ballerina/http;
      import ballerina/io;

      configurable string clientId = ?;
      configurable string clientSecret = ?;
      configurable string redirectUri = ?;
      configurable string code = ?;

      public function main() returns error? {
         http:Client hubspotClient = check new ("https://api.hubapi.com");

         // Construct the form-urlencoded payload
         string payload = string `grant_type=authorization_code&code=${code}&redirect_uri=${redirectUri}&client_id=${clientId}&client_secret=${clientSecret}`;

         http:Request tokenRequest = new;
         tokenRequest.setPayload(payload);
         tokenRequest.setHeader("Content-Type", "application/x-www-form-urlencoded");

         // Send POST request to fetch the access token
         http:Response tokenResponse = check hubspotClient->post("/oauth/v1/token", tokenRequest);

         // Parse and print the response
         json responseJson = check tokenResponse.getJsonPayload();
         io:println("Access and Refresh Token Response: ", responseJson);
      }

>    ```

Reviewer: Let's remove this part.

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

 - Linux/MacOS:
```bash
curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
```

 - Windows:
 ```bash
 curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token ^
  --header 'content-type: application/x-www-form-urlencoded' ^
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
 ```

 This command will return the access token and refresh token necessary for API calls.

 ```json
{
  "token_type": "bearer",
  "refresh_token": "<Refresh Token>",
  "access_token": "<Access Token>",
  "expires_in": 1800
}
 ```

 5. Store the access token securely for use in your application.

## Quickstart

To use the `HubSpot Transactional Emails` connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

> Import the ``hubspot.marketing.`transactional`` module and `oauth2` module.

Reviewer: ```suggestion
Import the `ballerinax/hubspot.marketing.'transactional` module and `ballerina/oauth2` module.
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

```ballerina

> import ballerinax/hubspot.marketing.`transactional as hsmtransactional;

Reviewer: ```suggestion
import ballerinax/hubspot.marketing.'transactional as hsmtransactional;
```

Author: done [39cebf8](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/39cebf8975ce224ea0f55eca10f5e74ea4d72ae2)

5. Use the obtained authorization code to run the following curl command, replacing <your_client_id>, <your_redirect_url>, and <your_authorization_code> with your specific values:

 - Linux/MacOS:
```bash
curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
```

 - Windows:
 ```bash
 curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token ^
  --header 'content-type: application/x-www-form-urlencoded' ^
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
 ```

 This command will return the access token and refresh token necessary for API calls.

 ```json
{
  "token_type": "bearer",
  "refresh_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "access_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "expires_in": 1800
}
 ```

 6. Store the access token securely for use in your application.

## Quickstart

> To use the `HubSpot Transactional emails` connector in your Ballerina application, update the `.bal` file as follows:

Reviewer 2: ```suggestion
To use the `HubSpot Transactional Emails` connector in your Ballerina application, update the `.bal` file as follows:
```

Author: done [4481219](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/44812198184b6debf4e6e5a4aa12d848509093db)

### Step 1: Import the module

Import the ``hubspot.marketing.`transactional`` module and `oauth2` module.

```ballerina
import ballerinax/hubspot.marketing.`transactional as hsmtransactional;
import ballerina/oauth2;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials obtained in the above steps as follows:

   ```toml
    clientId = <Client Id>
    clientSecret = <Client Secret>
    refreshToken = <Refresh Token>
   ```

2. Instantiate a `OAuth2RefreshTokenGrantConfig` with the obtained credentials and initialize the connector with it.

    ```ballerina
   configurable string clientId = ?;
   configurable string clientSecret = ?;
   configurable string refreshToken = ?;
   
   hsmtransactional:OAuth2RefreshTokenGrantConfig auth = {
       clientId,
       clientSecret,
       refreshToken,
       credentialBearer: oauth2:POST_BODY_BEARER
   };
   final hsmtransactional:Client hubSpotTransactional = check new ({ auth });
   ```

### Step 3: Invoke the connector operation

Now, utilize the available connector operations. A sample usecase is shown below.

#### Create a SMTP API Token

```ballerina

> SmtpApiTokenRequestEgg payload = {
>     createContact: true,
>     campaignName: "Campaign1"
> };
> SmtpApiTokenView response = check hubSpotTransactional->/smtp\-tokens.post(payload);

Reviewer 2: ```suggestion
hsmtransactional:SmtpApiTokenRequestEgg payload = {
    createContact: true,
    campaignName: "Campaign1"
};
hsmtransactional:SmtpApiTokenView response = check hubSpotTransactional->/smtp\-tokens.post(payload);
```

Author: done [4481219](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/44812198184b6debf4e6e5a4aa12d848509093db)

5. Use the obtained authorization code to run the following curl command, replacing <your_client_id>, <your_redirect_url>, and <your_authorization_code> with your specific values:

 - Linux/MacOS:
```bash
curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
```

 - Windows:
 ```bash
 curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token ^
  --header 'content-type: application/x-www-form-urlencoded' ^
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
 ```

 This command will return the access token and refresh token necessary for API calls.

 ```json
{
  "token_type": "bearer",
  "refresh_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "access_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "expires_in": 1800
}
 ```

 6. Store the access token securely for use in your application.

## Quickstart

> [//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
> 

Reviewer 2: No need for this text as we have already added the relevant details. Please check for other places as well.
```suggestion
```

```ballerina
import ballerinax/hubspot.marketing.transactional;
import ballerina/oauth2;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials obtained in the above steps as follows:

   ```toml
    clientId = <Client Id>
    clientSecret = <Client Secret>
    refreshToken = <Refresh Token>
   ```

2. Instantiate a `OAuth2RefreshTokenGrantConfig` with the obtained credentials and initialize the connector with it.

    ```ballerina
   configurable string clientId = ?;
   configurable string clientSecret = ?;
   configurable string refreshToken = ?;

   ConnectionConfig config = {
      auth: {
         clientId,
         clientSecret,
         refreshToken,
         credentialBearer: oauth2:POST_BODY_BEARER
      }
   };

   final Client hubSpotClient = check new (config);
   ```

### Step 3: Invoke the connector operation

Now, utilize the available connector operations. A sample usecase is shown below.

#### Create a SMTP API Token

```ballerina
SmtpApiTokenRequestEgg payload = {
        createContact: true,
        campaignName: "Campaign1"
    };

> SmtpApiTokenView|error response = check base_client->/smtp\-tokens.post(payload,{});

Reviewer 2: ```suggestion
SmtpApiTokenView response = check base_client->/smtp\-tokens.post(payload,{});
```

Author: done [bf032cb](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/bf032cb175f45f79addd76a87333365c7b046402)

> SmtpApiTokenView|error response = check base_client->/smtp\-tokens.post(payload,{});

Reviewer 2: Do we need to pass an empty `{}` to the client action ? If not lets remove that.

Author: done [bb8fede](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/bb8fede918c052530072b597b1949b46788cf800)

Author: If I did not pass it is showing error

```

> 
> 
> 
> 

Reviewer 2: ```suggestion

```

Author: done [bb8fede](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/bb8fede918c052530072b597b1949b46788cf800)

> [//]: # (TODO: Add overview mentioning the purpose of the module, supported REST API versions, and other high-level details.)

Reviewer 2: ```suggestion
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

"[HubSpot](https://www.hubspot.com/) is an AI-powered customer relationship management (CRM) platform. 

> The ``ballerinax/hubspot.marketing.`transactional`` offers APIs to connect and interact with the [marketing.transactional API](https://developers.hubspot.com/docs/reference/api/marketing/emails/transactional-emails) endpoints, specifically based on the [API guide](https://developers.hubspot.com/docs/guides/api/marketing/emails/transactional-emails)"

Reviewer 2: ```suggestion
The ``ballerinax/hubspot.marketing.`transactional`` offers APIs to connect and interact with the [Hubspot Marketing Transactional Emails API](https://developers.hubspot.com/docs/reference/api/marketing/emails/transactional-emails) endpoints, specifically based on the [HubSpot REST API](https://developers.hubspot.com/docs/reference/api/overview).
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

## Setup guide

> [//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)

Reviewer 2: ```suggestion
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

> To use the HubSpot connector, you must have access to the HubSpot API through a [HubSpot developer account](https://developers.hubspot.com/get-started) and a project under it. If you do not have a HubSpot Developer account, you can sign up for one [here](https://app.hubspot.com/signup-hubspot/developers?_ga=2.207749649.2047916093.1734412948-232493525.1734412948&step=landing_page).

Reviewer 2: ```suggestion
To use the HubSpot Marketing Transactional Emails connector, you must have access to the HubSpot API through a HubSpot developer account and a HubSpot App under it. Therefore you need to register for a developer account at HubSpot if you don't have one already.
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

1. Open the [HubSpot Developer Portal](https://app.hubspot.com/login/?loginRedirectUrl=https%3A%2F%2Fapp.hubspot.com%2Fsignup-hubspot%2Fdevelopers%3F_ga%3D2.269102326.624948025.1734413225-1764281074.1734413225%26step%3Dlanding_page)

2. Click on the 'App' tab and select an existing project or create a new project for which you want API keys and Authentication Access. 

![alt text](<docs/setup/resources/hubspot-app-home.png>)

In order to create a new poject, you must provide a public app name and a redirect url/s. Optionally you can add a app logo and a description for the app.

![alt text](docs/setup/resources/hubspot-new-app-info.png)

To add redirect url/s for the app, click the 'Auth' tab on top of the page, and navigate to 'Redirect URLs' section.

![alt text](docs/setup/resources/hubspot-new-app-auth.png)

![alt text](docs/setup/resources/hubspot-new-app-redirect-url.png)

### Setp 2. Obtain Client ID and Client Secret.

1. After completing the project setup, you will be provided with your client id and client secret.  Make sure to save the provided client id and client secret.

![alt text](docs/setup/resources/hubspot-new-app-client-id-secret.png)

> ### 2. Setup OAuth 2.0 Flow

Reviewer 2: Does this falls as a subtopic under `Setp 2. Obtain Client ID and Client Secret.` topic ? If so I think we have to adjust formatting.
```suggestion
#### 2. Setup OAuth 2.0 Flow
```

Author: done [704576d](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/704576d36ec14098cd1cbd4e5f442da4f56c7171)

5. Use the obtained authorization code to run the following curl command, replacing <your_client_id>, <your_redirect_url>, and <your_authorization_code> with your specific values:

 - Linux/MacOS:
```bash
curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token \
  --header 'content-type: application/x-www-form-urlencoded' \
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
```

 - Windows:
 ```bash
 curl --request POST \
  --url https://api.hubapi.com/oauth/v1/token ^
  --header 'content-type: application/x-www-form-urlencoded' ^
  --data 'grant_type=authorization_code&code=<your_authorization_code>&redirect_uri=<your_redirect_url>&client_id=<your_client_id>&client_secret=<your_client_secret>'
 ```

 This command will return the access token and refresh token necessary for API calls.

 ```json
{
  "token_type": "bearer",
  "refresh_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "access_token": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "expires_in": 1800
}
 ```

 6. Store the access token securely for use in your application.

> 
> 

Reviewer 2: ```suggestion

```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

## Quickstart

To use the `HubSpot Transactional emails` connector in your Ballerina application, update the `.bal` file as follows:

### Step 1: Import the module

Import the ``hubspot.marketing.`transactional`` module and `oauth2` module.

```ballerina

> import ballerinax/hubspot.marketing.`transactional;

Reviewer 2: ```suggestion
import ballerinax/hubspot.marketing.`transactional as hsmtransactional;
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

import ballerina/oauth2;
```

### Step 2: Instantiate a new connector

1. Create a `Config.toml` file and, configure the obtained credentials obtained in the above steps as follows:

   ```toml
    clientId = <Client Id>
    clientSecret = <Client Secret>
    refreshToken = <Refresh Token>
   ```

2. Instantiate a `OAuth2RefreshTokenGrantConfig` with the obtained credentials and initialize the connector with it.

    ```ballerina

>    configurable string clientId = ?;
>    configurable string clientSecret = ?;
>    configurable string refreshToken = ?;
> 
>    ConnectionConfig config = {
>       auth: {
>          clientId,
>          clientSecret,
>          refreshToken,
>          credentialBearer: oauth2:POST_BODY_BEARER
>       }
>    };
> 
>    final Client hubSpotClient = check new (config);

Reviewer 2: ```suggestion
   configurable string clientId = ?;
   configurable string clientSecret = ?;
   configurable string refreshToken = ?;
   
   hsmtransactional:OAuth2RefreshTokenGrantConfig auth = {
       clientId,
       clientSecret,
       refreshToken,
       credentialBearer: oauth2:POST_BODY_BEARER
   };
   final hsmtransactional:Client hubSpotTransactional = check new ({ auth });
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

   ```

### Step 3: Invoke the connector operation

Now, utilize the available connector operations. A sample usecase is shown below.

#### Create a SMTP API Token

```ballerina

> SmtpApiTokenRequestEgg payload = {
>         createContact: true,
>         campaignName: "Campaign1"
>     };

Reviewer 2: ```suggestion
SmtpApiTokenRequestEgg payload = {
    createContact: true,
    campaignName: "Campaign1"
};
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

> SmtpApiTokenView response = check base_client->/smtp\-tokens.post(payload,{});

Reviewer 2: I think this should work.
```suggestion
SmtpApiTokenView response = check hubSpotTransactional->/smtp\-tokens.post(payload);
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

```
## Examples

The `Ballerina HubSpot Marketing Transactional Connector` connector provides practical examples illustrating usage in various scenarios. Explore these [examples](https://github.com/module-ballerinax-hubspot.marketing.transactional/tree/main/examples/), covering the following use cases:

> 1. [Manage Tokens](/examples/manage-tokens) - see how the Hubspot API can be used to create SMTP API token and query it.

Reviewer 2: Shall we follow the approach used in Ballerina zendesk connector when linking the examples in the documentation ?

https://github.com/ballerina-platform/module-ballerinax-zendesk?tab=readme-ov-file#examples

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)

> [//]: # (TODO: Add examples)

Reviewer 2: ```suggestion
```

Author: done [8690be1](https://github.com/ballerina-platform/module-ballerinax-hubspot.marketing.transactional/pull/3/commits/8690be120e49b602c1525636ce018f6d69e6ddc7)
